exporters/logging-otlp/src/main/java/io/opentelemetry/exporter/logging/otlp/internal/logs/OtlpStdoutLogRecordExporterProvider.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.exporter.logging.otlp.internal.logs;
+
+import io.opentelemetry.sdk.autoconfigure.spi.ConfigProperties;
+import io.opentelemetry.sdk.autoconfigure.spi.logs.ConfigurableLogRecordExporterProvider;
+import io.opentelemetry.sdk.logs.export.LogRecordExporter;
+
+/**
+ * {@link LogRecordExporter} SPI implementation for {@link OtlpStdoutLogRecordExporter}.
+ *
+ * <p>This class is internal and is hence not for public use. Its APIs are unstable and can change
+ * at any time.
+ */
+public class OtlpStdoutLogRecordExporterProvider implements ConfigurableLogRecordExporterProvider {
+  @Override
+  public LogRecordExporter createExporter(ConfigProperties config) {
+    OtlpJsonLoggingLogRecordExporterBuilder builder = OtlpStdoutLogRecordExporter.builder();
+    return builder.build();
+  }
+
+  @Override
+  public String getName() {
+    return "otlp-stdout";
```

Comment:
I think we discussed in the meeting that this is the right way for adding new experimental enum values.
Not sure who said it, but I can find out.

Reply:
answered by @jsuereth on https://cloud-native.slack.com/archives/C01N7PP1THC/p1725025067755049?thread_ts=1725009130.383539&cid=C01N7PP1THC

https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/versioning-and-stability.md#semantic-conventions-stability
<- See how lists of well known values can be added to

also within semconv, there's a stabilty attribute per-value of an eum:
https://github.com/open-telemetry/semantic-conventions/blob/main/model/registry/http.yaml#L38

Reply:
Hmm.. this isn't a semantic convention though.

Let's think this through:
- `opentelemetry-sdk-extension-autoconfigure` is stable. 
- Stable means we can't the public API can't change.
- Stable also means that we can't go and change the behavior of properties. For example, we could go and say that we now interpret the value of `otel.exporter.otlp.timeout` as seconds instead of milliseconds.
- Similarly, we can't say that setting `otel.metrics.exporter=otlp` suddenly uses the http/protobuf exporter instead of grpc. 
- Here, we're considering adding a new exporter value, such that setting `otel.logs.exporter=otlp-stdout` results in the `OtlpStdoutLogRecordExporter` being configured. If `otlp-stdout` is removed or changes names, is it a breaking change for stable `opentelemetry-sdk-extension-autoconfigure` module?
- The answer depends on how you interpret autoconfigure's boundaries. The `otlp-stdout` value doesn't actually occur anywhere in the autoconfigure code - its only loaded via SPI. And so the stable autoconfigure module can always load SPI implementations which make breaking changes according. For example, extension components from contrib can make breaking changes to their config properties - does that mean that autoconfigure had a breaking change? Of course not.
- This is a tricky case because the SPI implementation is part of the core `opentelemetry-java` project where we have strict standards around compatibility, and because the value is currently being proposed / debated in specification, and will be subject to change for some period of time.

I'm inclined to say that we should include `experimental` in the name to make it clear to users that the value is subject to change. I.e. something like `experimental-otlp-stdout`.

Reply:
I don't think we need `experimental` in the name if users have to bring in an `-alpha` artifact to use it (e.g. similar to prometheus exporter).

I do think we need `experimental` in the name if users don't have to bring in an `-alpha` artifact to use it (the case as it is today in the PR).

Reply:
I like the idea of alpha artifact as this is going to be experimental for a whilei

Reply:
I see you've moved the `OtlpStdoutLogRecordExporterComponentProvider`, `OtlpStdoutLogRecordExporterProvider` to `opentelemetry-sdk-extension-incubator`.

This is more confusing than my suggestion:

> I think we should keep it in the opentelemetry-exporter-logging-otlp artifact, and include experimental-* in the autoconfigure name used for OTEL_{SIGNAL}_EXPORTER.

Java agent users get the `opentelemetry-sdk-extension-incubator` artifact bundled automatically so never have to opt-in to and acknowledge that the component is experimental. We also don't have any precedent for SPIs implementations being packaged separately from the artifact they implement.

Keeping `experimental-*` in the name is the simplest to convey to users that this is subject to change. We should prioritize making the spec stable so we can drop the prefix rather than jumping through hoops to make this component appear more stable than it actually is.